windows/power_operation_scripts/set_power_script.yml

```diff
@@ -0,0 +1,145 @@
+# Copyright 2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Description:
+#   Set and check soft power operation
+#
+- name: "Start to test {{ power_script_type }} {{ power_script_op }} script"
+  ansible.builtin.debug:
+    msg:
+      - "Start to test {{ power_script_type }} {{ power_script_op }} script"
+
+- name: "Set {{ power_script_type }} {{ power_script_op }} script"
+  include: set_power_script_{{ power_script_type }}.yml
+  vars:
+    power_cmd_op: "{{ power_script_op }}"
+
+- name: "Check the current configuration for {{ power_script_op }} script"
+  include_tasks: ../utils/win_execute_cmd.yml
+  vars:
+    win_powershell_cmd: "& '{{ vmware_toolbox_cmd_path }}' script {{ power_script_op }} current"
+
+- name: "Print the current configuration for {{ power_script_op }} script"
+  ansible.builtin.debug: var=win_powershell_cmd_output
+
+- name: "Check current configuration is expected"
+  ansible.builtin.assert:
+    that:
+      - script_file_path
+      - win_powershell_cmd_output.stdout is defined
+      - win_powershell_cmd_output.stdout.rstrip() == script_file_path
+    fail_msg: "The {{ power_script_op }} script is {{ win_powershell_cmd_output.stdout | default('') }}, not expected {{ script_file_path }}."
+    success_msg: "The {{ power_script_op }} script is set to the expected one {{ win_powershell_cmd_output.stdout | default('') }}."
+
+- name: "Get content of file tools.conf"
```

Comment:
You can consider using https://docs.ansible.com/ansible/latest/collections/ansible/builtin/ini_lookup.html to lookup key's value if file content is too long to print in log.